src/StorageSync/StorageSync/RegisteredServer/RegisterServerCommand.cs

```diff
@@ -0,0 +1,195 @@
+        [Alias(StorageSyncAliases.StorageSyncServiceIdAlias)]
+        public string ParentResourceId { get; set; }
+
+
+        [Parameter(Mandatory = false, HelpMessage = HelpMessages.AsJobParameter)]
+        public SwitchParameter AsJob { get; set; }
+
+        protected override string Target => StorageSyncServiceName ?? ParentObject?.StorageSyncServiceName ?? ParentResourceId;
+
+        protected override string ActionMessage => $"Create a new Registered Server for Storage Sync Service {StorageSyncServiceName ?? ParentObject?.StorageSyncServiceName ?? ParentResourceId}";
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Target, ActionMessage))
+            {
+                base.ExecuteCmdlet();
+
+                ExecuteClientAction(() =>
+                {
+                    var parentResourceIdentifier = default(ResourceIdentifier);
+
+                    if (!string.IsNullOrEmpty(ParentResourceId))
+                    {
+                        parentResourceIdentifier = new ResourceIdentifier(ParentResourceId);
+
+                        if (!string.Equals(StorageSyncConstants.StorageSyncServiceType, parentResourceIdentifier.ResourceType, System.StringComparison.OrdinalIgnoreCase))
+                        {
+                            throw new PSArgumentException($"Invalid Argument {nameof(ParentResourceId)}", nameof(ParentResourceId));
+                        }
+                    }
+
+                    var resourceGroupName = ResourceGroupName ?? ParentObject?.ResourceGroupName ?? parentResourceIdentifier?.ResourceGroupName;
+
+                    if (string.IsNullOrEmpty(resourceGroupName))
+                    {
+                        throw new PSArgumentException($"Invalid Argument {nameof(ResourceGroupName)}", nameof(ResourceGroupName));
+                    }
+
+                    var parentResourceName = StorageSyncServiceName ?? ParentObject?.StorageSyncServiceName ?? parentResourceIdentifier?.ResourceName;
+
+                    if (string.IsNullOrEmpty(parentResourceName))
+                    {
+                        throw new PSArgumentException($"Invalid Argument {nameof(StorageSyncServiceName)}", nameof(StorageSyncServiceName));
+                    }
+
+                    if (!SubscriptionId.HasValue)
+                    {
+                        throw new PSArgumentException("No subscription found", nameof(SubscriptionId));
+                    }
+
+                    RegisteredServer resource = PerformServerRegistration(resourceGroupName, SubscriptionId.Value, parentResourceName);
+
+                    WriteObject(resource);
+                });
+            }
+        }
+        private RegisteredServer PerformServerRegistration(string resourceGroupName, Guid subscriptionId, string storageSyncServiceName)
+        {
+            if (subscriptionId == null)
+            {
+                throw new PSArgumentException("No subscription found");
```

Comment:
resource string.  Also, please include a reommended action to correct the error for the user in the exception message